core/Cargo.toml

```diff
@@ -250,8 +250,7 @@ reqwest = { version = "0.11.18", features = [
 ], default-features = false }
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
-# TODO: remove `rt` after https://github.com/apache/opendal/issues/3948
-tokio = { version = "1.27", features = ["sync", "rt"] }
+tokio = { version = "1.27", features = ["sync"] }
```

Comment:
if our upstream dependencies include `rt`, it may still detect this feature.

Reply:
But I suppose compiling "BlockingLayer" should fail for core crate itself. You can see that it still compiles ...

Reply:
It depends on the method of feature solving, and it does not always occur according to natural logic.

Reply:
Yeah. We depend on reqwest which hard depends on tokio runtime ..

Reply:
[Feature unification](https://doc.rust-lang.org/nightly/cargo/reference/features.html#feature-unification) in Cargo is addition and there is no way to opt-out if any dependency in your dependency closure activating a feature.

There is a pre-RFC might get this resolved, and we'd like to see people drive it: <https://internals.rust-lang.org/t/pre-rfc-mutually-excusive-global-features/19618>.

Reply:
> if our upstream dependencies include `rt`, it may still detect this feature.

This PR is still meaningful that we are not the blocker for other projects to remove `tokio/rt` anymore.